src/twisted/internet/test/test_fdset.py

```diff
@@ -115,6 +115,19 @@ def fail():
         # Getting here means the right thing happened probably.
 
 
+    def test_removeReader_selectables_nokey(self):
+        """
+        L{reactor.removeReader()} accepts an L{IReadDescriptor} provider
+        previously passed to C{reactor.addReader()}. Checks if it doesn't
+        raise KeyError exception
+        """
+        reactor, fd, server = self._simpleSetup()
+        reactor.addReader(fd)
+        reactor._selectables = {}
```

Comment:
I have the same problem and it seems to be related to `twisted.internet.threads.deferToThread()` - at least for me. I created a .rpy script that triggers this exception after several connection attempts (see https://gist.github.com/DefaultUser/f8bd6b1cb60fc260fd397113878bebeb). It takes a while till it actually happens, so you need to spam F5 for a while